src/CommunityToolkit/Xamarin.CommunityToolkit/Behaviors/EventToCommandBehaviorGeneric.shared.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+
+namespace Xamarin.CommunityToolkit.Behaviors
+{
+	/// <summary>
+	/// This <see cref="EventToCommandBehavior"/> cast the sender object to a specific type defined by the user. 
+	/// </summary>
+	/// <typeparam name="TType">The type that you want to receive in your <see cref="Xamarin.Forms.Command"/> </typeparam>
+	public sealed class EventToCommandBehavior<TType> : EventToCommandBehavior
+	{
+		protected override void OnTriggerHandled(object sender = null, object eventArgs = null)
+		{
+			var parameter = CommandParameter
+				?? EventArgsConverter?.Convert(eventArgs, typeof(object), null, null)
+				?? eventArgs;
+
+			if (parameter?.GetType() != typeof(TType))
+			{
+				// nulling it to avoid a cast exception
+				parameter = null;
+				parameter = Convert.ChangeType(parameter, typeof(TType));
```

Comment:
in this case, TType should be a reference type, right?

Reply:
it looks like parameter is always null here

Reply:
No, isn't the parameter will be the type of `eventArgs`

```csharp
var parameter = CommandParameter
?? EventArgsConverter?.Convert(eventArgs, typeof(object), null, null)
?? eventArgs;
```